### DISCUSSION TITLE
How to throw validation error for a nullable field?

### DISCUSSION THREAD

**Discussion by Cavallando** [MichalLytek/type-graphql#1203] [Q&A]
Right now it appears that having a validation on a nullable field results in the the field being set to `undefined` as opposed the the validation error being thrown. From a user perspective, I would expect the error to be thrown so that I have some action to take as opposed to no error being thrown and the user being left confused as to why the field I input was not accepted. Am I missing something from the docs? Is this supported?

Example:

```
@InputType()
export class UpdateAddressInput {
  @Field({ nullable: true })
  @Length(2, 2, {
    message: "State must be given as two letter format (ie. NY, NJ, etc)",
  })
  state?: string;
}

const result = await query(
  queryString, {
    where: { id: address.id },
      data: {
        state: chance.state({ full: true }),
      },
    },
);
```

**Comment by MichalLytek** [comment]
Nullable + validation = (null/undefined) or the correct value

If you want to throw error on null, just make the field non-nullable, so GraphQL won't let you skip that field or pass null.
